include/dxc/DXIL/DxilResourceProperties.h

```diff
@@ -25,26 +25,30 @@ struct DxilResourceProperties {
   DXIL::ResourceKind  Kind;
   static constexpr unsigned kSampleCountUndefined = 0x7;
 
+  struct DxilTyped {
+    DXIL::ComponentType CompType : 5; // TypedBuffer/Image.
```

Comment:
GCC is keep complaining 5-bit is too small for ComponentType. It's a GCC bug, which is fixed in 8.4 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61414). Unfortunately this warning doesn't have a -W code, no way to turn it off.

Reply:
That should go away with upcoming changes to the struct where this will be a uint8_t instead of a 5-bit field anyways.